### PR TITLE
fix bug in circuit_breaker.go about SlowRequestRatio

### PR DESCRIPTION
**Pull request by Jayice-zjw**
### Describe what this PR does / why we need it
when slowRequestRatio was set to 1.0d, requests will **never** be blocked. Because the currentRatio will never be bigger than 100%

### Describe how you did it
Special treatment for case when slowRequestRatio was set to 1.0d

### Special notes for reviews

### CODE REVIEW

**Review by sanxun0325 — COMMENTED** [review]
Maybe the same problem exists in errorRatioCircuitBreaker

**Comment by Jayice-zjw** [comment]
> Maybe the same problem exists in errorRatioCircuitBreaker

Exactly. Thanks for your tip

**Comment by louyuting** [comment]
The Rule.Threshold means the max value that circuit breaker allowed. If the statistic ratio is great than Threshold, the circuit breaker will open.

From my perspective, setting Rule.Threshold as 1.0 doesn't make sense. May be we should check (Threshold==1.0) in https://github.com/alibaba/sentinel-golang/blob/51b28fdb3acd42bf0f47ac1ac00af2be237f2f55/core/circuitbreaker/rule_manager.go#L347

**Comment by Jayice-zjw** [comment]
> The Rule.Threshold means the max value that circuit breaker allowed. If the statistic ratio is great than Threshold, the circuit breaker will open.
> 
> From my perspective, setting Rule.Threshold as 1.0 doesn't make sense. May be we should check (Threshold==1.0) in
> 
> https://github.com/alibaba/sentinel-golang/blob/51b28fdb3acd42bf0f47ac1ac00af2be237f2f55/core/circuitbreaker/rule_manager.go#L347
You make a good point! But I think it would be better to support this, in case this feature is needed in some scenarios? What do you think?

File: core/circuitbreaker/circuit_breaker.go [hunk 1]
```diff
@@ -290,7 +290,8 @@ func (b *slowRtCircuitBreaker) OnRequestComplete(rt uint64, err error) {
 		return
 	}
 
-	if slowRatio > b.maxSlowRequestRatio {
+	// Special case when maxSlowRequestRatio is set to 1.0 should be handled
```
Comment (louyuting) [inline core/circuitbreaker/circuit_breaker.go:293]: Maybe we need to change checking logic.
Using "slowRatio >= b.maxSlowRequestRatio"、”errorRatio >= b.errorRatioThreshold“ and ”errorCount >= b.errorCountThreshold“ 

It's easier to understand.

**Comment by louyuting** [comment]
> > The Rule.Threshold means the max value that circuit breaker allowed. If the statistic ratio is great than Threshold, the circuit breaker will open.
> > From my perspective, setting Rule.Threshold as 1.0 doesn't make sense. May be we should check (Threshold==1.0) in
> > https://github.com/alibaba/sentinel-golang/blob/51b28fdb3acd42bf0f47ac1ac00af2be237f2f55/core/circuitbreaker/rule_manager.go#L347
> > 
> > You make a good point! But I think it would be better to support this, in case this feature is needed in some scenarios? What do you think?

Maybe we need improve the checking logic here. Refer to my comments

**Comment by louyuting** [comment]
Maybe  you do a wrong update. Below is my comments:
>Maybe we need to change checking logic.
Using "slowRatio >= b.maxSlowRequestRatio"、”errorRatio >= b.errorRatioThreshold“ and ”errorCount >= b.errorCountThreshold“
It's easier to understand.

**Comment by Jayice-zjw** [comment]
> Maybe you do a wrong update. Below is my comments:
> 
> > Maybe we need to change checking logic.
> > Using "slowRatio >= b.maxSlowRequestRatio"、”errorRatio >= b.errorRatioThreshold“ and ”errorCount >= b.errorCountThreshold“
> > It's easier to understand.

Modified.Thanks for suggestions

**Review by louyuting — APPROVED** [review]
Nice work and looking forward more~

**Comment by Jayice-zjw** [comment]
> Nice work and looking forward more~

appreciated for your help

**Pull request merged**